nmdc_api_utilities/minter.py

```diff
@@ -12,14 +12,20 @@
 class Minter(NMDCSearch):
     """
     Class to interact with the NMDC API to mint new identifiers.
+
+    Parameters
+    ----------
+    auth : NMDCAuth
+        An instance of the NMDCAuth class for authentication.
     """
 
-    def __init__(self, env="prod"):
+    def __init__(self, env="prod", auth: NMDCAuth = None):
+        self.env = env
+        self.auth = auth or NMDCAuth()
         super().__init__(env=env)
 
-    def mint(
-        self, nmdc_type: str, client_id: str, client_secret: str, count: int = 1
-    ) -> str | list[str]:
+    @requires_auth
+    def mint(self, nmdc_type: str, count: int = 1) -> str | list[str]:
```

Comment:
I _don't think_ that username / password authentication will work with the minter :(. And we should either keep backwards compatibility by allowing for directly passing`client_id: str, client_secret: str` (my preference) or we'll need to change all the calls to this in the `mass_spectrometry_metadata_gneration` repo.